Tests/ApodiniTests/SharedSemanticModelBuilderTests.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  SharedSemanticModelBuilderTests.swift
+//  
+//
+//  Created by Lorena Schlesinger on 06.12.20.
+//
+
+import XCTest
+import Vapor
+@testable import Apodini
+
+final class SharedSemanticModelBuilderTests: XCTestCase {
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    var app: Application!
+    
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        app = Application(.testing)
+    }
+    
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+        let app = try XCTUnwrap(self.app)
+        app.shutdown()
+    }
+    
+    struct TestHandler: Component {
+        @Parameter
+        var name: String
+        
+        func handle() -> String {
+            "Hello \(name)"
+        }
+    }
+    
+    struct TestHandler2: Component {
+        @Parameter
+        var name: String
+        
+        @Parameter("someId", .http(.path))
+        var id: Int
+        
+        func handle() -> String {
+            "Hello \(name)"
+        }
+    }
+    
+    struct TestComponent: Component {
+        @PathParameter
+        var name: String
+        
+        var content: some Component {
+            Group("a") {
```

Comment:
I would also suggest to add a test case where there is a `Group` and some other component in the `Group("a")` just to make sure we also cover this edge case.

Reply:
I assume this is solved via a0f36a5 by @lschlesinger

Reply:
yes, just added this corner case!